src/core/hle/kernel/k_scoped_lock.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+// This file references various implementation details from Atmosphere, an open-source firmware for
+// the Nintendo Switch. Copyright 2018-2020 Atmosphere-NX.
+
+#pragma once
+
+#include "common/common_types.h"
+
+namespace Kernel {
+
+template <typename T>
+concept KLockable = !std::is_reference<T>::value && requires(T & t) {
+    { t.Lock() }
+    ->std::same_as<void>;
+    { t.Unlock() }
+    ->std::same_as<void>;
+};
+
+template <typename T>
+requires KLockable<T> class KScopedLock : NonCopyable {
```

Comment:
```suggestion
requires KLockable<T> class KScopedLock {
```

I'd opt for either explicitly deleting and defining both the copy and move operations within the class for consistency. Currently the move operations are implicitly not defined due to the presence of a destructor, so it makes for better readability to be explicit about the intent of both of them.